src/ScintillaWnd.cpp

```diff
@@ -913,18 +899,13 @@ void CScintillaWnd::RestoreCurrentPos(const CPosData& pos)
     }
     Call(SCI_CHOOSECARETX);
 
+    Call(SCI_ENSUREVISIBLE, pos.m_nFirstVisibleLine);
```

Comment:
here, I get a complete hang: scintilla loops indefinitely on the fourth restored file (of 9 files to open on startup)
The problem is that SCI_ENSUREVISIBLE requires the view to paint, but that doesn't happen.

Reply:
I'm not expert enough with scintilla to understand this, but it doesn't surprise me it would be an affect which is why my comment says it needs testing thoroughly. Hopefully you can fix this particular issue without losing the fixes my patch aims to address.

Reply:
the problem here is that you "fixed" your flicker problem by forcing Scintilla to paint every document in full right after loading it. This causes an immense slowdown when you have bigger files or complex lexers.
So I'm sorry to say that this is not a fix I can approve - the slowdown is just too big (waiting 5 minutes is really unacceptable, and that's just for one big logfile - imagine I had more logfiles open or even my 300MB generated c++ file which has a much more complex lexer)

Reply:
You're making it sound like bowpad must draw the screen at page 1 and then again at the right place in order to prevent the slowdown. If that was actually the case every editor that used scintilla would have this problem and they don't, so the problem must be fixable.

By far the bulk of my patch is to ensure the position information is available to Bowpad early enough to use at the right time. So that means you can take everything in my patch except the changes to ScntillaWnd.cpp. So I suggest you take all of the files minus ScntillaWnd.cpp, and then patch the previous version of ScintillaWnd.cpp prior to my version, to your liking, so that it uses the position information that's now available to it to avoid the problem. That shouldn't be much work for you to finish it off.  There's no way you have to draw the screen twice to do this. What's left for you to do should be a small change for you, maybe one line or two of code now.

Reply:
Oh I forgot to say: can you please confirm you actually tested your slowdown test case on the version I've submitted and that your slowdown comments isn't just based on your memory of your prior code. I just want to be sure it still applies. But whether you patch my ScntillaWnd.cpp or the prior version, I'm sure there can't be much left to fix the drawing problem now the location information is available at the right time. Check the comment in the Centre function, you can perhaps conditionally call SCI_ENSUREVISIBLE like it does to avoid the extra call. That might be enough. Whatever the issue is that's remaining, I'm sure it isn't much.

Reply:
It might be worth mentioning what your slowdown test case is too so I can try it on my machine. I don't really doubt there's an issue there, just that fixing it is a small patch to my patch if anything and you're better placed than me to know what that change is.

Reply:
just use a very big file that needs a lexer. Around 500MB big should suffice to show you a significant delay when loading that file on startup (i.e. using a session).
I have here a log file (*.log) that's 900MB big.
And yes, I've tested this, otherwise I wouldn't have noticed the delay/slowdown.

Reply:
I created a 300mb test file by appending src\Commands\*.cpp to the test file other over and over until the test file size was 300mb.  The test file contained 7.6 million lines of c++.

I then compared the time taken to load that test file by the last released version of bowpad, against the time taken to load that same file with a version of bowpad patched with my original patch in this pull request.

The tl;dr is: my additional one line patch makes both versions load at the same time of about 1.45 minutes. So unless you find anything conceptually wrong with it, please apply this change. Without this second patch one line patch,  the version without the extra 1 line patch takes about 30 seconds longer.

Reply:
I still get a slowdown, not as much but still noticeable. But I can overlook that.
However, #134 was fixed and with this PR it stops working again.

Reply:
Hmm, I explicitly tested that situation and was sure I found it worked. But you're right, I just tested it again and it's back to not working. I'll take a look again.